streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -314,9 +341,29 @@ public synchronized final TopologyBuilder addSource(String name, String... topic
      * @return this builder instance so methods can be chained together; never null
      */
     public synchronized final TopologyBuilder addSource(String name, Pattern topicPattern) {
-        return addSource(name, (Deserializer) null, (Deserializer) null, topicPattern);
+        return addSource(name, null, (Deserializer) null, (Deserializer) null, topicPattern);
+    }
+
+    /**
+     * Add a new source that consumes from topics matching the given pattern
+     * and forward the records to child processor and/or sink nodes.
+     * The source will use the {@link org.apache.kafka.streams.StreamsConfig#KEY_SERDE_CLASS_CONFIG default key deserializer} and
+     * {@link org.apache.kafka.streams.StreamsConfig#VALUE_SERDE_CLASS_CONFIG default value deserializer} specified in the
+     * {@link org.apache.kafka.streams.StreamsConfig stream configuration}.
+     *
+     * @param name the unique name of the source used to reference this node when
+     * @param offsetReset the offset reset value for this source acceptable values earliest or latest.  For this parameter
```

Comment:
Just talked to @guozhangwang about this -- in fact, `auto.offset.reset` should be set to `none` within the consumer all the time, as we do the reset handling "manually" within Streams now (instead of relying on Kafka's built-in reset logic).

Thus, the setting of `AUTO_OFFSET_RESET_CONFIG` can be set to any value the user wants, but we do not use this value to configure the `KafkaConsumer` -- `KafkaConsumer` always get value `none` for `auto.offset.reset`. Thus, the try-catch logic in place, triggers all the time when there is no valid committed offset found and we apply a topic-specific reset policy (if there is any) or the user specified (via `AUTO_OFFSET_RESET_CONFIG`) as fallback.